test/fiddle/test_pointer.rb

```diff
@@ -170,6 +186,31 @@ def test_free=
       assert_equal free.ptr, ptr.free.ptr
     end
 
+    def test_free!
+      ptr = Pointer.malloc(4)
+      refute ptr.freed?
+      ptr.free!
+      assert ptr.freed?
+      ptr.free!                 # you can safely run it again
+      assert ptr.freed?
```

Comment:
What do we think `free!` and `freed?` should do when there's no free function attached?

Reply:
How about doing nothing?

```ruby
refute ptr.freed?
ptr.free!
refute ptr.freed?
ptr.free!
refute ptr.freed?
```

Reply:
Yes that's what we do at the moment. As long as everyone else also thinks that's fine I'll do that.

Reply:
Done.